docs: initial content for Testing in NativeScript (#15)
NativeScript/docs: content/guide/testing.md

---
title: Testing
---

::: warning Note
Be sure you have prepare/built/run the app at least once before starting the unit test runner.
:::

For more information about end-to-end testing, see [`@nativescript/detox` plugin](https://docs.nativescript.org/plugins/detox.html).

> When you develop new features inside your app, you can ensure that they are working properly and that past functionality has not regressed by writing and executing unit tests on a regular basis. With the NativeScript CLI, you can write and execute unit tests using [Jasmine](http://jasmine.github.io/), [Mocha](https://mochajs.org/) with [Chai](http://chaijs.com/) or [QUnit](https://qunitjs.com/).

Reviewer: ```suggestion
Regularly writing and executing unit tests using tools like [Jasmine](http://jasmine.github.io/), [Mocha](https://mochajs.org/) with [Chai](http://chaijs.com/), or [QUnit](https://qunitjs.com/) through the NativeScript CLI helps ensure proper functioning of new features and prevents regressions in existing functionality during app development.
```

To run your unit tests, the NativeScript CLI uses [Karma](http://karma-runner.github.io/latest/index.html).

### Before You Begin

> Before writing and running unit tests, verify that you have completed the following steps.

Reviewer: ```suggestion
Before writing and running unit tests, the completion of the following steps must be verified.
```
